src/builder.rs

```diff
@@ -307,6 +303,12 @@ impl Builder {
                     .take(num_actions - num_recipients),
             );
 
+            // Shuffle the spends and recipients, so that learning the position of a
+            // specific spent note or output note doesn't reveal anything on its own about
+            // the meaning of that note in the transaction context.
+            self.spends.shuffle(&mut rng);
+            self.recipients.shuffle(&mut rng);
```

Comment:
I verified that the [algorithm used in rand 0.8.4](https://github.com/rust-random/rand/blob/0.8.4/src/seq/mod.rs#L586-L592) is a correctly implemented Durstenfeld shuffle, using `rng.gen_range` which is specified to be uniform. (The [implementation of that](https://github.com/rust-random/rand/blob/0.8.4/src/distributions/uniform.rs) is complicated but ends up using rejection sampling.)